### PR TITLE
Charts: Format percentage values with smart decimal handling

### PR DESCRIPTION
**Pull request by annacmc**
## Proposed changes:
* Add `formatPercentage()` utility function that intelligently formats percentages by removing trailing zeros and capping at 2 decimal places
* Update pie chart and semi-circle pie chart legends to use smart percentage formatting (e.g., "30%" instead of "30.0%")
* Update conversion funnel charts to use consistent smart percentage formatting across main rate, step rates, and tooltips
* This improves readability by showing "30%", "30.1%", "30.25%" instead of always showing one decimal place

### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Does this pull request change what data or activity we track or use?

No, it does not. 

## Testing instructions:

* Go to Storybook -> Charts -> PieChart 
* Verify that pie chart legends display percentages without unnecessary decimals (e.g., "60%" not "60.0%")
* Go to Storybook -> Charts -> PieSemiCircleChart
* Verify the same smart percentage formatting in semi-circle pie chart legends
* Go to Storybook -> Charts -> ConversionFunnelChart
* Verify that funnel chart percentages display smartly: whole numbers as "100%", decimals preserved when meaningful "45.2%"
* Test edge cases: very small percentages (0.05%), large percentages (99.99%), and verify rounding works correctly
* Ensure no visual regressions in any chart rendering

### CODE REVIEW

File: projects/js-packages/charts/src/components/conversion-funnel-chart/test/conversion-funnel-chart.test.tsx [hunk 1]
```diff
@@ -47,7 +48,7 @@ describe( 'ConversionFunnelChart', () => {
 			renderWithoutTheme( <ConversionFunnelChart { ...defaultProps } /> );
 
 			// Check main rate is displayed (first occurrence)
-			expect( screen.getAllByText( '10.3%' ) ).toHaveLength( 2 ); // Main rate + Purchase step
+			expect( screen.getAllByText( formatPercentage( 10.3 ) ) ).toHaveLength( 2 ); // Main rate + Purchase step
```
Comment (kangzj) [inline projects/js-packages/charts/src/components/conversion-funnel-chart/test/conversion-funnel-chart.test.tsx]: Are these changes here intentional? I actually would prefer them as literal here in these tests.

File: projects/js-packages/charts/src/utils/format-metric-value.ts [hunk 2]
```diff
@@ -90,3 +90,15 @@ export const formatMetricValue = (
...
 };
+
+/**
+ * Format a percentage value with smart decimal handling.
+ * Removes unnecessary trailing zeros and caps at 2 decimal places.
+ *
+ * @param value - The percentage value (0-100 range)
+ * @return Formatted percentage string (e.g., "30%", "30.1%", "30.25%")
+ */
+export const formatPercentage = ( value: number ): string => {
```
Comment (kangzj) [inline projects/js-packages/charts/src/utils/format-metric-value.ts]: We should probably reuse the percentage formatting from `@automattic/number-formatters`

Comment (kangzj) [reply projects/js-packages/charts/src/utils/format-metric-value.ts]: Also, let's create separate files for all new public functions so that those files wouldn't grow infinitely.

File: projects/js-packages/charts/src/components/conversion-funnel-chart/test/conversion-funnel-chart.test.tsx (hunk 1 above)
Comment (annacmc) [reply projects/js-packages/charts/src/components/conversion-funnel-chart/test/conversion-funnel-chart.test.tsx]: No worries, I can easily fix this back to literal 👍 😄

**Review by kangzj — APPROVED** [review]
🚀
